File: content-solutions/binary-search/cut-cables.js
function solution(vines, k) {
  let lo = 1;
  let hi = Math.max(...vines);
  let answer = 0;
  while (lo <= hi) {
    const mid = Math.floor((lo + hi) / 2);
    let pieces = 0;
    for (const v of vines) pieces += Math.floor(v / mid);
    if (pieces >= k) {
      answer = mid;
      lo = mid + 1;
    } else hi = mid - 1;
  }
  return answer;
}

File: content-solutions/binary-search/cut-cables.java
class Solution {
    public int solution(int[] vines, int k) {
        long lo = 1, hi = 0, answer = 0;
        for (int v : vines) hi = Math.max(hi, v);
        while (lo <= hi) {
            long mid = (lo + hi) / 2;
            long pieces = 0;
            for (int v : vines) pieces += v / mid;
            if (pieces >= k) {
                answer = mid;
                lo = mid + 1;
            } else hi = mid - 1;
        }
        return (int) answer;
    }
}
